toolchain/check/import_ref.cpp

```diff
@@ -1371,7 +1372,7 @@ static auto AddAssociatedEntities(ImportContext& context,
           context.import_entity_names().Get(import_ref->entity_name_id).name_id;
     } else {
       CARBON_FATAL("Unhandled associated entity kind: {0}",
-                   context.import_insts().Get(inst_id).kind());
+                   context.import_insts().GetWithAttachedType(inst_id).kind());
```

Comment:
I would prefer to not make this change. We can see locally that we're not inspecting the type of the instruction here, only its `kind()`, so getting an attached type is unnecessary. Generally `GetWithAttachedType` is a weird thing to do, so I think we should only be calling it when we have a reason.